codebreaker_smn/spec/codebreaker_smn/game_spec.rb

```diff
@@ -0,0 +1,159 @@
+require 'spec_helper'
+
+RSpec.describe CodebreakerSmn::Game do
+  let(:game) { described_class.new }
+  let(:game_code) { [1, 2, 3, 4] }
+  let(:win_guess) { '1234' }
+  let(:invalid_guesses) { ['', 'zz', '123', '11234', '7890', Object] }
+
+  before do
+    game.new_game
+    game.start
+    game.username = 'valid_username'
+    game.difficulty = 'easy'
+  end
+
+  context 'with #start' do
+    it 'generates secret code' do
+      expect(game.instance_variable_get(:@code)).not_to be_empty
+    end
+
+    it 'saves 4 numbers secret code' do
+      expect(game.instance_variable_get(:@code).size).to eq(4)
+    end
+
+    it 'saves secret code with numbers from 1 to 6' do
+      expect(game.instance_variable_get(:@code).join).to match(/[1-6]+/)
+    end
+  end
+
+  context 'with #guess_code' do
+    context 'when validates user guess' do
+      it 'with valid guess' do
+        game.instance_variable_set(:@code, game_code)
+        expect(game.guess_code(win_guess)).to eq('++++')
+      end
+
+      it 'with invalid guess' do
+        game.instance_variable_set(:@code, game_code)
+        invalid_guesses.each do |invalid_guess|
+          expect(game.guess_code(invalid_guess)).to be_nil
+        end
+      end
+    end
+
+    context 'when processes user guesses' do
+      let(:examples_1) do
```

Comment:
let's use all the examples from the Technical requirements

Reply:
I'm using all the examples from Technical requirements, I just divided them into examples 1, 2, 3, 4 because of different secret code